connectors/src/api/webhooks/webhook_slack.ts

```diff
@@ -371,7 +377,86 @@ const _webhookSlackAPIHandler = async (
         }
         break;
       }
+      case "channel_created": {
+        const teamId = req.body.team_id;
+        const slackChannelId = req.body.event?.channel;
+        const slackConfiguration = await SlackConfigurationModel.findOne({
+          where: {
+            slackTeamId: teamId,
+          },
+        });
+        if (!slackConfiguration || !slackChannelId) {
+          return apiError(req, res, {
+            api_error: {
+              type: "slack_configuration_not_found",
+              message: `Slack configuration not found for teamId ${teamId}`,
+            },
+            status_code: 404,
+          });
+        }
+        const connectorId = slackConfiguration.connectorId;
 
+        const connector = await ConnectorResource.fetchById(connectorId);
+        if (!connector) {
+          return apiError(req, res, {
+            api_error: {
+              type: "connector_not_found",
+              message: `Connector ${req.params.connector_id} not found`,
+            },
+            status_code: 404,
+          });
+        }
+        const slackClient = await getSlackClient(connectorId);
+        const remoteChannel = await slackClient.conversations.info({
+          channel: slackChannelId,
+        });
+        const remoteChannelName = remoteChannel.channel?.name;
+
+        if (!remoteChannel.ok || !remoteChannelName) {
+          logger.error({
+            connectorId,
+            channelId: slackChannelId,
+            error: remoteChannel.error,
+          });
+          return new Err(
```

Comment:
Are we sure about this here? Don't we need to use `apiError` to actually return an error to the caller?

Reply:
👌🏼